File: erp/src/main/java/com/erp/model/Attendance.java
package com.erp.model;

import javax.persistence.*;
import java.util.Date;

@Entity(name = "attendance")
@Table(name = "attendance")
public class Attendance {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "attendance_code")
    private Long attendanceCode;

    @Column(name = "employee_code")
    private Long employeeCode;

    @Column(name = "employee_name")
    private String employeeName;

    @Column(name = "date")
    @Temporal(TemporalType.DATE)
    private Date date;

    @Column(name = "in_time")
    private String inTime;

    @Column(name = "out_time")
    private String outTime;

    @Column(name = "status")
    private String status;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getAttendanceCode() {
        return attendanceCode;
    }

    public void setAttendanceCode(Long attendanceCode) {
        this.attendanceCode = attendanceCode;
    }

    public Long getEmployeeCode() {
        return employeeCode;
    }

    public void setEmployeeCode(Long employeeCode) {
        this.employeeCode = employeeCode;
    }

    public String getEmployeeName() {
        return employeeName;
    }

    public void setEmployeeName(String employeeName) {
        this.employeeName = employeeName;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getInTime() {
        return inTime;
    }

    public void setInTime(String inTime) {
        this.inTime = inTime;
    }

    public String getOutTime() {
        return outTime;
    }

    public void setOutTime(String outTime) {
        this.outTime = outTime;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
    
}

File: ebs_flutter/lib/model/Attendance.dart
class Attendance {
  final int? id;
  final int? attendanceCode;
  final int employeeCode;
  final String? employeeName;
  final DateTime? date;
  final String? inTime;
  final String? outTime;
  final String? status;

  Attendance({
    this.id,
    this.attendanceCode,
    required this.employeeCode,
    this.employeeName,
    this.date,
    this.inTime,
    this.outTime,
    this.status,
  });

  factory Attendance.fromJson(Map<String, dynamic> json) {
    return Attendance(
      id: json['id'],
      attendanceCode: json['attendanceCode'],
      employeeCode: json['employeeCode'],
      employeeName: json['employeeName'],
      date: json['date'] != null ? DateTime.parse(json['date']) : null,
      inTime: json['inTime'],
      outTime: json['outTime'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'attendanceCode': attendanceCode,
      'employeeCode': employeeCode,
      'employeeName': employeeName,
      'date': date?.toIso8601String(),
      'inTime': inTime,
      'outTime': outTime,
      'status': status,
    };
  }
}